beancount.el

```diff
@@ -221,8 +221,14 @@ from the open directive for the relevant account."
   (concat "^\\(" beancount-date-regexp "\\) +"
           "\\(" (regexp-opt beancount-timestamped-directive-names) "\\) +"))
 
+(defconst beancount-metadata-base-regexp
+  "\\([a-z][A-Za-z0-9_-]+:\\)\\s-+\\(.+\\)")
+
 (defconst beancount-metadata-regexp
-  "^\\s-+\\([a-z][A-Za-z0-9_-]+:\\)\\s-+\\(.+\\)")
+  (concat "^\\s-+" beancount-metadata-base-regexp))
+
+(defconst beancount-metadata-indent-regexp
+  (concat "\\s-*" beancount-metadata-base-regexp))
```

Comment:
The important distinction between the two regexps is `+` being changed to `*`, i.e. allowing the case where a metadata line is not yet indented, and I think that sharing the common part of the two regexps is important for avoiding code repetition (particularly of a somewhat opaque "raw" regular expression rather than something that already builds upon other defined regexp constants).

Reply:
A metadata line not starting with whitespace is not valid Beancount syntax. I understand being lenient with what you accept as input, but a line needs to be drawn somewhere. I am sure we should not be accepting invalid Beancount syntax and try to make sense of it.

Reply:
See my response above, but additionally:

>  I understand being lenient with what you accept as input, but a line needs to be drawn somewhere.

The line, IMO, should be drawn after "ignoring a given line's *existing* indentation when determining how it *should* be indented". This allows us to unambiguously determine the user's intent and make editing their journal more natural and intuitive rather than fighting them for seemingly no reason at all.

What benefit comes from not being lenient in this one particular case?

> I am sure we should not be accepting invalid Beancount syntax and try to make sense of it.

As mentioned in the above response, `beancount-mode` *already does this*.

In addition to the given example about `beancount-mode` accepting and making sense of the invalid Beancount syntax of non-indented postings, `beancount-mode` assumes any line that begins with a number is a timestamped directive, so given a line
```
2
```
which is clearly invalid Beancount syntax, `beancount-mode` considers it to be a valid timestamped directive, at least as far as its indentation computation logic goes.